src/main/java/Parser.java

```diff
@@ -0,0 +1,102 @@
+import java.io.IOException;
+
+public class Parser {
+    private static final String SAVED_FORMAT_DELIMITER = " \\| ";
+    private final TaskList taskList;
```

Comment:
Consider placing TaskList instance outside of the Parser as parsing tend to only involve the translation process from a string into a specific object, in this case, a Task. The collection of tasks should be external.

Reply:
I agree with Tomashiwa on this, pass the taskList into Parser from your TaskList class will make it simpler.